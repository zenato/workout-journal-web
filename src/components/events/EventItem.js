import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import styles from './EventItem.scss';

const cx = classNames.bind(styles);

const EventItem = ({ item, onDetail }) => {
  const handleDetail = (e) => {
    e.preventDefault();
    onDetail(item.id);
  };
  return (
    <li key={item.id} className={cx('event-item')}>
      <a href="" onClick={handleDetail}>
        <div className={cx('info')}>
          <span className={cx('name')}>{item.name}</span>
          {item.value && (
            <span className={cx('value-unit')}>{item.value}{item.unit}</span>
          )}
          <div className={cx('remark')}>{item.remark}</div>
        </div>
      </a>
    </li>
  );
}

EventItem.propTypes = {
  item: PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    value: PropTypes.number,
    unit: PropTypes.string.isRequired,
    remark: PropTypes.string,
  }),
  onDetail: PropTypes.func.isRequired,
};

export default EventItem;
