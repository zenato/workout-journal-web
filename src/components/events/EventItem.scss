@import 'common';

.event-item {
  border: solid 1px #c7c7c7;
  margin-bottom: 5px;
  background-color: #fff;

  >a {
    padding: 10px;
    display: flex;

    &:hover {
      background-color: #f9f9f9;
    }
    &:active {
      background-color: #ebebeb;
    }
  }

  .info {
    flex: 1;
  }

  .name {
    font-weight: bold;
    color: #ff5008;
  }

  .value-unit {
    margin-left: 5px;
    color: #4e4e4e;
  }

  .remark {
    color: #a0a0a0;
    font-size: 0.7rem;
    margin-top: 5px;
  }
}
