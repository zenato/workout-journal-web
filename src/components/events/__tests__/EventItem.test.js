import React from 'react';
import { shallow } from 'enzyme';
import EventItem from '../EventItem';

describe('EventItem', () => {
  const props = {
    item: {
      id: '1',
      name: 'test name',
      value: null,
      unit: 'KG',
      remark: 'test mock data',
    },
    onDetail: jest.fn(),
  };

  it('should render value and unit if not empty `value` props.', () => {
    const component = shallow(<EventItem {...props} value={10} />);
    expect(component).toMatchSnapshot()
  });

  it('should call `onDetail` if click link.', () => {
    const component = shallow(<EventItem {...props} />);
    component.find('a').simulate('click', { preventDefault: () => {} });
    expect(props.onDetail).toBeCalled();
  });
});
