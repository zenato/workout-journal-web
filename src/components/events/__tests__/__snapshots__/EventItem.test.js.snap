// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventItem should render value and unit if not empty \`value\` props. 1`] = `
<li
  className="event-item"
  key="1"
>
  <a
    href=""
    onClick={[Function]}
  >
    <div
      className="info"
    >
      <span
        className="name"
      >
        test name
      </span>
      <div
        className="remark"
      >
        test mock data
      </div>
    </div>
  </a>
</li>
`;
